src/adapter/src/coord/dataflows.rs

```diff
@@ -556,53 +556,11 @@ impl<'a> DataflowBuilder<'a> {
         memo: &mut BTreeMap<GlobalId, bool>,
     ) -> Result<bool, RecursionLimitError> {
         self.checked_recur(|_| {
-            match self.catalog.get_entry(&id).item() {
-                CatalogItem::Source(source) => Ok(self.monotonic_source(source)),
-                CatalogItem::View(View { optimized_expr, .. })
-                | CatalogItem::MaterializedView(MaterializedView { optimized_expr, .. }) => {
-                    let mut view_expr = optimized_expr.clone().into_inner();
```

Comment:
IIUC we are just removing that code, right? I need to double-check that this does introduce optimizer regressions for materialized views.